Packs/Box/Integrations/BoxV2/BoxV2.py

```diff
@@ -1503,30 +1492,71 @@ def list_users_command(client: Client, args: Dict[str, Any]) -> CommandResults:
     )
 
 
-def upload_file_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+@polling_function(
+    'box-upload-file',
+    interval=arg_to_number(demisto.args().get('retry_after') or 30),
+    poll_message='Uploading file to Box'
+)
+def upload_file_command(args: Dict[str, Any], client: Client) -> PollResult:
     """
     Uploads a file to Box. For files with a size over the set limit, the file will be uploaded in
     chunks. For files which are under the limit, they will be submitted using a POST request.
 
+    :param args: demisto.args() dictionary - Used to pass the necessary arguments to the client function.
     :param client: Client - Initialized Client object.
-    :param args: demisto.args() dictionary - Used to pass the necessary arguments to the client
-    function.
     :return: CommandResults - Returns a CommandResults object which is consumed by the
     return_results function in main()
     """
     entry_id: str = args.get('entry_id')  # type:ignore
     file_name: str = args.get('file_name')  # type:ignore
     folder_id: str = args.get('folder_id')  # type:ignore
     as_user: str = args.get('as_user')  # type:ignore
-    response: dict = client.upload_file(entry_id=entry_id, file_name=file_name, folder_id=folder_id,
-                                        as_user=as_user)
-    readable_output = "File was successfully uploaded"
-    return CommandResults(
-        readable_output=readable_output,
-        outputs_prefix='Box.File',
-        outputs_key_field='id',
-        outputs=response.get('entities')
-    )
+
+    maximum_chunk_size = 20000000
+    demisto_file_object = demisto.getFilePath(entry_id)
+    file_name = file_name or demisto_file_object.get('name')
+    file_path = demisto_file_object.get('path')
+    file_size = os.path.getsize(file_path)
+    #  Box requires files to have a file extension. We validate that here.
+    if '.' not in file_name:  # type: ignore
```

Comment:
This function could be more robust. Right now if someone uploads a hidden file without an extension it will pass this validation and ultimately fail.

```python
def has_valid_file_extension(filename):
    name, ext = os.path.splitext(filename)
    # Ensure that the extension is not empty and that the name part is not empty.
    return ext != '' and ext != '.' and len(name) > 0
```

We may also have a stronger validation somewhere in CSP 🤷